caddyhttp/maxrequestbody/maxrequestbody.go

```diff
@@ -0,0 +1,59 @@
+package maxrequestbody
+
+import (
+	"regexp"
+	"strconv"
+
+	"github.com/mholt/caddy"
+	"github.com/mholt/caddy/caddyhttp/httpserver"
+)
+
+var SIZE_RE = regexp.MustCompile("[1-9][0-9]*[kmKM]?")
```

Comment:
To prevent maintenance headaches in the future, can we not use a regular expression? :) This is a simple job for parsing. If it's a positive integer, use it as bytes. If it's a string suffixed with any case variation of KB, MB, (or GB, I suppose), split it out and parse the non-negative number before it.